File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/fragments/presenters/RecentPresenter.java
package com.tevinjeffrey.vapor.ui.files.fragments.presenters;

import com.squareup.otto.Bus;
import com.squareup.otto.Subscribe;
import com.tevinjeffrey.vapor.events.DatabaseUpdateEvent;
import com.tevinjeffrey.vapor.okcloudapp.model.CloudAppItem;

import javax.inject.Inject;

public class RecentPresenter extends BaseFilesPresenterImpl {
    public RecentPresenter() {
        itemType = CloudAppItem.ItemType.ALL;
    }

    @Inject
    Bus bus;

    @Subscribe
    public void dbUpdate(DatabaseUpdateEvent event) {
        loadData(false, false, false);
    }

    @Override
    public void onPause() {
        super.onPause();
        bus.unregister(this);
    }

    @Override
    public void onResume() {
        super.onResume();
        bus.register(this);
    }

}

File: vapor-app/src/main/java/com/tevinjeffrey/vapor/ui/files/fragments/presenters/RecentPresenter.kt
package com.tevinjeffrey.vapor.ui.files.fragments.presenters

import com.squareup.otto.Bus
import com.squareup.otto.Subscribe
import com.tevinjeffrey.vapor.events.DatabaseUpdateEvent
import com.tevinjeffrey.vapor.events.RefreshEvent
import com.tevinjeffrey.vapor.events.UploadEvent
import com.tevinjeffrey.vapor.okcloudapp.model.CloudAppItem

import javax.inject.Inject

class RecentPresenter : BaseFilesPresenterImpl() {
    init {
        itemType = CloudAppItem.ItemType.ALL
    }

    @Subscribe
    fun dbUpdate(event: DatabaseUpdateEvent) {
        loadData(false, false, true)
    }

    @Subscribe
    fun onUploadEvent(event: UploadEvent) {
        loadData(false, true, false)
    }

    @Subscribe
    fun onRefreshEvent(event: RefreshEvent) {
        if (view != null && view!!.isVisibleInPager) {
            loadData(true, true, false)
        }
    }

    override fun onPause() {
        super.onPause()
        bus.unregister(this)
    }

    override fun onResume() {
        super.onResume()
        bus.register(this)
    }

}
